subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestingIntegrationTest.groovy

```diff
@@ -541,4 +542,66 @@ class TestingIntegrationTest extends JUnitMultiVersionIntegrationSpec {
         result.testClass("UsefulNPETest")
             .testFailed("testFailingGetMessage", equalTo('Could not determine failure message for exception of type UsefulNPETest$1: java.lang.RuntimeException'))
     }
+
+    @IgnoreIf({ GradleContextualExecuter.embedded })
```

Comment:
Out of curiosity: why does this not work with the embedded executor?

Reply:
Because serialization only happens if we don't use the embedded runner (everything is happening within the same process)